src/store/movieStore.ts

```diff
@@ -0,0 +1,64 @@
+import {  ref, computed } from 'vue';
+import { defineStore } from 'pinia';
+import { IMockMovie, mockMoviesList } from "../mockedData/mockedData.ts";
+import { useSearch } from '../utils.ts';
+
+const useMovieStore = defineStore(
+    'movieStore', () => {
+        const initialList = ref(mockMoviesList);
+        const sortByParam = ref("releaseDate");
+        const isMovieDetailOpen = ref<number | null>(null);
+        const { searchQuery, setSearchQuery, searchedMovies, setFilterParam, searchMovies } = useSearch(initialList.value);
+
+        const setSortByParam = (param: string) => {
+            sortByParam.value = param;
+        }
+
+        const toggleMovieDetailPanel = (val: number | null) => {
+            isMovieDetailOpen.value = val;
+            if(typeof val === 'number'){
+                setFilterParam("genres");
+            }
+            else{
+                setFilterParam("title");
+                setSearchQuery("");   
+            }
+        }
+
+        const getMovieById = computed(() =>{
+            if(isMovieDetailOpen.value){
+                const movie = initialList.value.find((movie: IMockMovie) => movie.id === isMovieDetailOpen.value);
+                movie ? setSearchQuery(movie.genres[0]) : undefined;
```

Comment:
Conditional assignment use for returning value or assigning a value.
Use `if` condition for this.

Even better, if you will have a watcher for the `selectedMovie`: https://vuejs.org/guide/essentials/watchers.html